src/openforms/formio/components/custom.py

```diff
@@ -243,21 +254,33 @@ def build_serializer_field(
 
     @staticmethod
     def as_json_schema(component: MapComponent) -> JSONObject:
-        label = component.get("label", "Map coordinate")
+        label = component.get("label", "Map")
+        interactions = component["interactions"]
 
-        base = {
+        properties = [
+            {
+                "properties": {
+                    "type": {"type": "string", "const": geometry_type},
+                    "coordinates": deepcopy(GEO_JSON_COORDINATE_SCHEMAS[geometry_type]),
+                },
+                "additionalProperties": False,
+            }
+            for geometry_type in GeoJsonGeometryTypes.values
+            # Only include the schema of types that are allowed
+            if interactions.get(GEO_JSON_TYPE_TO_INTERACTION[geometry_type], True)
```

Comment:
```suggestion
            if interactions.get(GEO_JSON_TYPE_TO_INTERACTION[geometry_type], False)
```
If the interaction key is not defined, it'll be disabled in the frontend, so the fallback value should also be falsy.